services/client/services_test.go

```diff
@@ -136,6 +136,16 @@ func TestAdvertiseErrors(t *testing.T) {
 	err = sd.Advertise(ad)
 	require.NoError(t, err)
 
+	hbGen.Lock()
+	hb := hbGen.hbs[serviceKey(sid)]
+	hbGen.Unlock()
+
+	// hb store should show advertisement (almost) immediately
+	time.Sleep(time.Millisecond)
```

Comment:
Increased the relevant times to give the tests a bit more headroom. Ran with `-c 100` and didn't see any flakes so far.